ml-agents-envs/mlagents/envs/base_env.py

```diff
@@ -0,0 +1,308 @@
+"""
+Python Environment API for the ML-Agents toolkit
+The aim of this API is to expose groups of similar Agents evolving in Unity
+to perform reinforcement learning on.
+There can be multiple groups of similar Agents (same observations and actions
+spaces) in the simulation. These groups are identified by a agent_group that
+corresponds to a single group of Agents in the simulation.
+For performance reasons, the data of each group of agents is processed in a
+batched manner. When retrieving the state of a group of Agents, said state
+contains the data for the whole group. Agents in these groups are identified
+by a unique int identifier that allows tracking of Agents across simulation
+steps. Note that there is no guarantee that the number or order of the Agents
+in the state will be consistent across simulation steps.
+A simulation steps corresponds to moving the simulation forward until at least
+one agent in the simulation sends its observations to Python again. Since
+Agents can request decisions at different frequencies, a simulation step does
+not necessarily correspond to a fixed simulation time increment.
+Changes from ML-Agents v0.11 :
```

Comment:
Minor but is this a good place for a changelist?  I worry it will become out of date quickly

Reply:
I put this here because it is major change but I can remove it.
I have not tested on windows, will do.

Reply:
- [x] windows

Reply:
![Screen Shot 2019-12-09 at 2 35 45 PM](https://user-images.githubusercontent.com/28320361/70478985-3fbaac80-1a91-11ea-8a88-c15107bb49a1.png)
Looks like it is working